### ISSUE TITLE
[TF2] [Feature request] Let us equip more than three cosmetic items at once, and loosen restrictions on clipping

### ISSUE THREAD

**Issue by chexo3** [ValveSoftware/Source-1-Games#4660]
Some really cool outfits are actually impossible to equip in-game because of the three item limit. In addition, some items can look good together even if they’re clipping through each other. While it may be pertinent to show a warning when there are clipping issues, I don’t think it’s necessary in $CURRENT_YEAR with how diverse the game’s art style and cosmetics has gotten.

That said, there may be framerate concerns if players equip a large amount of cosmetics, so there probably _should_ be a limit, but it doesn’t need to be three. It could also be a more dynamic limit, based on how many triangles/polys are being rendered with a certain loadout. Maybe you could even save some of your cosmetic’s poly count budget by setting some to render at a lower LOD by default? 

Either way, I really love this game, ancient as it is, and I’d love to adorn my de facto main class Engineer with moar cosmetics.

If this sounds pointless or like a waste of time, I get it. But think about it this way: If I have a favorite outfit for a class, but I can’t add X item to it because of already wearing 3 things, I’m probably not going to buy X item from the Mann Co. Store or the community market, which means Valve loses out on a sale!

It’s not exactly a secret Valve hasn’t paid much attention to TF2 lately, but we have had some improvements with bot issues, not to mention bug fixes, localization updates (as much of a meme as those are, they still help folks and they’re not free), not to mention casual mode and Mann up servers still running. It’s clear this game is at least self-sufficient monetarily.

TL;DR loosen limits on number of cosmetics able to be worn and which can be combined. IMO I would buy more cosmetics as a result of a change like this.

**Comment by Ashetf2** [comment]
I start to wonder if this repo is getting filled up with issues that are not a "feature" per se, like this one or #4480

**Comment by AzureWoof** [comment]
An argument can be made for more cosmetic slots and updating certain equip regions to make more sense. But removing equip regions entirely is out of the question. Equip region restrictions are in place for a reason. Without them, we would have ways to completely cover up team colors over time, leading to a negative impact on gameplay-- or being able to completely cover up your head with a bunch of cosmetics, making it more difficult for Snipers to hit their shots.

**Comment by chexo3** [comment]
> I start to wonder if this repo is getting filled up with issues that are not a "feature" per se, like this one or #4480

What? How is this not a feature request?

**Comment by chexo3** [comment]
> An argument can be made for more cosmetic slots and updating certain equip regions to make more sense. But removing equip regions entirely is out of the question. Equip region restrictions are in place for a reason. Without them, we would have ways to completely cover up team colors over time, leading to a negative impact on gameplay-- or being able to completely cover up your head with a bunch of cosmetics, making it more difficult for Snipers to hit their shots.

Good point. I was thinking it’d probably be fine for cosmetics like coats and jackets that are usually team colored to have a button or badge cosmetic on top of them, for example. Or combining team colored cosmetics in general.

I don’t think the sniper issue is that much of a problem, it doesn’t change where items are equipped, just allows items to share an equip region. It won’t displace anything.

Maybe the game could do a simple test server-side, essentially rendering the player and their loadout at various distances and checking if they’re sufficiently team colored. Computer vision tools that could perform this test well already exist. So when you equip a loadout, it checks how much of your original player model is covered, and at a certain threshold triggers the color check.

**Comment by JoriKos** [comment]
> Replying to https://github.com/ValveSoftware/Source-1-Games/issues/4660#issuecomment-1370348445

Personally I think when you have to introduce third-party systems to check your loadouts, there's a big flaw in your suggestion and is a lot more work than you think.

**Comment by chexo3** [comment]
> > Replying to [#4660 (comment)](https://github.com/ValveSoftware/Source-1-Games/issues/4660#issuecomment-1370348445)
> 
> Personally I think when you have to introduce third-party systems to check your loadouts, there's a big flaw in your suggestion and is a lot more work than you think.

Where did I say it’s a third party system? What I’m saying is the code and techniques already exist. Libraries for it too. It wouldn’t be effortless, but unless someone at Valve feels like reinventing the wheel it’s definitely not hard.

**Comment by jh34ghu43gu** [comment]
Item servers are already dying, adding another slot on loadouts might just push it over the edge.

**Comment by matty4kek** [comment]
> I start to wonder if this repo is getting filled up with issues that are not a "feature" per se, like this one or #4480

This repo sadly started getting bombarded by "dumb" requests once it got popular after the -quickswitch fiasco.
It's sad to see, really. We went from having actual issues being posted on daily basis to unrealistic feature requests being spammed.

**Comment by chexo3** [comment]
> > I start to wonder if this repo is getting filled up with issues that are not a "feature" per se, like this one or #4480
> 
> This repo sadly started getting bombarded by "dumb" requests once it got popular after the -quickswitch fiasco.

Oh go on, act like I’m not here. 

I genuinely think this would be a cool thing to have.

**Comment by MikhailaTF** [comment]
> > > I start to wonder if this repo is getting filled up with issues that are not a "feature" per se, like this one or #4480
> > 
> > 
> > This repo sadly started getting bombarded by "dumb" requests once it got popular after the -quickswitch fiasco.
> 
> Oh go on, act like I’m not here.
> 
> I genuinely think this would be a cool thing to have.

And we genuinely disagree with you.

**Comment by JoriKos** [comment]
> Responding to https://github.com/ValveSoftware/Source-1-Games/issues/4660#issuecomment-1371368282

It's not that people personally dislike you for this or anything, but the issue people have with this is that if these issues stop being majority bug reports and start being half bug reports half "add this unused content" or something similar, this place will be a lot less desired for devs to look through if all that's being posted is feature requests.

**Comment by chexo3** [comment]
> > Responding to [#4660 (comment)](https://github.com/ValveSoftware/Source-1-Games/issues/4660#issuecomment-1371368282)
> 
> It's not that people personally dislike you for this or anything, but the issue people have with this is that if these issues stop being majority bug reports and start being half bug reports half "add this unused content" or something similar, this place will be a lot less desired for devs to look through if all that's being posted is feature requests.

It's annoying the way they're talking about it though. I thought this would be a fun addition to the game, but some folks came and shat all over it acting like I can't read their comments too.

**Comment by JoriKos** [comment]
> Responding to https://github.com/ValveSoftware/Source-1-Games/issues/4660#issuecomment-1399352160

You're bound to face criticism when you have an idea, especially one as wild as this one, but you also posted it at a time where there was an above average amount of people starting to ask for random cut content to be added to the game (because of the BLU Sydney Sleeper change). As these requests can easily flood the issues section and overshadow important bug reports, it would become a less valuable source for Valve for bugs. People aren't criticising your idea, but rather the influx of cut content requests.

**Comment by chexo3** [comment]
I still think this is a decent idea. Giving players ways to wear more cosmetics and ways to minimize clipping when doing so would let folks be more creative. I do think there should be some sort of LOD system though, to save lower end computer's resources.

**Comment by matty4kek** [comment]
> I'd like more cosmetic slots, but they shouldn't just generally loosen restrictions it would have to be specific and for specific cosmetics not to mention how TF2 works is very complicated
> <img alt="Image" width="850" height="432" src="https://private-user-images.githubusercontent.com/258138127/619754637-adb0d9be-2ed8-40be-bcd2-bb4c6496ee73.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODM2NzExNDQsIm5iZiI6MTc4MzY3MDg0NCwicGF0aCI6Ii8yNTgxMzgxMjcvNjE5NzU0NjM3LWFkYjBkOWJlLTJlZDgtNDBiZS1iY2QyLWJiNGM2NDk2ZWU3My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNzEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDcxMFQwODA3MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZDBiMTNiODkxNjkyNGUxMTk2ZjNiYzkxZTk3ZDBlODU3ZTRiMWFhNjljMTlkODM4ODZlNTZiMzQxYjNhNzhkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.AIPm21QCjJXW3ycWbnwDiXCEEJ2so1Bl3R2H2NVVF2I">

I think Valve avoids loosening up restrictions on hats like in picture you've posted because at that point it would make the hat as a "misc" and as far as I know they want to avoid putting more miscs in the game.
At least for hats that can be obtained as Unusual.
